Tools/Scripts/libraries/webkitscmpy/webkitscmpy/pull_request.py

```diff
@@ -60,6 +60,23 @@ def __repr__(self):
                 self.content,
             )
 
+    class Status(object):
+        def __init__(self, url, status, name=None, description=None):
+            self.url = url
+
+            if status not in ("error", "failure", "pending", "success"):
```

Comment:
This follows from what GitHub does, and seems to map nicely to what BitBucket does (["Valid values: STOPPED, FAILED, INPROGRESS, SUCCESSFUL"](https://developer.atlassian.com/cloud/bitbucket/rest/api-group-pullrequests/#api-repositories-workspace-repo-slug-pullrequests-pull-request-id-statuses-get)). That said, I'm a bit confused on the BitBucket side—all our code seems to use `pull-requests` in the URL, whereas all the BitBucket documentation uses `pullrequests`. Am I looking at the wrong documentation?

Reply:
Oh… Oh. Bitbucket Server and Bitbucket have… different REST APIs.

Reply:
https://developer.atlassian.com/server/bitbucket/rest/v811/api-group-builds-and-deployments/#api-api-latest-projects-projectkey-repos-repositoryslug-commits-commitid-builds-get I think is what we want.